hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestClientRollback.java

```diff
@@ -175,9 +187,78 @@ public void testSavepointAndRollback() throws Exception {
 
       dataFiles = partitionPaths.stream().flatMap(s -> view3.getAllBaseFiles(s).filter(f -> f.getCommitTime().equals("004"))).collect(Collectors.toList());
       assertEquals(0, dataFiles.size(), "The data files for commit 004 should be rolled back");
+
+      if (testFailedRestore) {
+        //test to make sure that restore commit is reused when the restore fails and is re-ran
+        HoodieInstant inst =  table.getActiveTimeline().getRestoreTimeline().getInstants().get(0);
+        String restoreFile = table.getMetaClient().getBasePathV2().toString() + "/.hoodie/" +  inst.getFileName();
+        File commitFile = new File(restoreFile);
+        commitFile.delete();
+        if (!failedRestoreInflight) {
+          File inflightFile = new File(restoreFile + ".inflight");
+          inflightFile.delete();
+        }
+        try (SparkRDDWriteClient newClient = getHoodieWriteClient(cfg)) {
+          //restore again
+          newClient.restoreToSavepoint(savepoint.getTimestamp());
+          metaClient = HoodieTableMetaClient.reload(metaClient);
+          table = HoodieSparkTable.create(getConfig(), context, metaClient);
+          //verify that there is only 1 restore commit
+          assertEquals(1, table.getActiveTimeline().getRestoreTimeline().getInstants().size());
```

Comment:
lets also verify 
that its complete. 
and matches the first time restore instant time